devicetypes/fibargroup/fibaro-flood-sensor-zw5.src/fibaro-flood-sensor-zw5.groovy

```diff
@@ -1,300 +1,531 @@
+    }
+
+    cmds = cmds + encapSequence(cmdsGet, 1000)
+    cmds << "delay " + (5000 + cmdCount * 1500)
+    cmds << encap(zwave.wakeUpV1.wakeUpNoMoreInformation())
+    results = results + response(cmds)
+
+    return results
+}
+
+
+def syncCheck() {
+    logging("${device.displayName} - Executing syncCheck()", "info")
+    def notSynced = []
+    def count = 0
+
+    if (device.currentValue("syncStatus") != "synced") {
+        parameterMap().each {
+            if (state."$it.key"?.state == "notSynced") {
+                notSynced << it
+                logging "Sync failed! Verify parameter: ${notSynced[0].num}"
+                logging "Sync $it.key " + state."$it.key"
+                sendEvent(name: "batteryStatus", value: "Sync incomplited! Check parameter nr. ${notSynced[0].num}")
+                count = count + 1
+            }
+        }
+    }
+    if (count == 0) {
+        logging("${device.displayName} - Sync Complete", "info")
+        sendEvent(name: "syncStatus", value: "synced")
+    } else {
+        logging("${device.displayName} Sync Incomplete", "info")
+        if (device.currentValue("syncStatus") != "failed") {
+            sendEvent(name: "syncStatus", value: "incomplete")
+        }
+    }
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.manufacturerspecificv2.ManufacturerSpecificReport cmd) {
+    log.debug "ManufacturerSpecificReport"
+    log.debug "manufacturerId:   ${cmd.manufacturerId}"
+    log.debug "manufacturerName: ${cmd.manufacturerName}"
+    log.debug "productId:        ${cmd.productId}"
+    log.debug "productTypeId:    ${cmd.productTypeId}"
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.manufacturerspecificv2.DeviceSpecificReport cmd) {
+    log.debug "DeviceSpecificReport"
+    log.debug "deviceIdData:                ${cmd.deviceIdData}"
+    log.debug "deviceIdDataFormat:          ${cmd.deviceIdDataFormat}"
+    log.debug "deviceIdDataLengthIndicator: ${cmd.deviceIdDataLengthIndicator}"
+    log.debug "deviceIdType:                ${cmd.deviceIdType}"
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.versionv1.VersionReport cmd) {
+    log.debug "VersionReport"
+    log.debug "applicationVersion:      ${cmd.applicationVersion}"
+    log.debug "applicationSubVersion:   ${cmd.applicationSubVersion}"
+    log.debug "zWaveLibraryType:        ${cmd.zWaveLibraryType}"
+    log.debug "zWaveProtocolVersion:    ${cmd.zWaveProtocolVersion}"
+    log.debug "zWaveProtocolSubVersion: ${cmd.zWaveProtocolSubVersion}"
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.batteryv1.BatteryReport cmd) {
+    log.debug "BatteryReport"
+    log.debug "cmd: "+cmd
+    log.debug "location: "+location
+    def timeDate = location.timeZone ? new Date().format("yyyy MMM dd EEE h:mm:ss a", location.timeZone) : new Date().format("yyyy MMM dd EEE h:mm:ss")
+    //def timeDate = new Date().format("yyyy MMM dd EEE HH:mm:ss", location.timeZone)
+    sendEvent(name: "battery", value: cmd.batteryLevel)

```

Comment:
there's old logic in the handler that has been removed that needs to treat the special z-wave value of 255 as "low battery"